src/Plugin/Action/SystemPageRedirect.php

```diff
@@ -0,0 +1,141 @@
+ *       description = @Translation("Whether to append a destination parameter to the URL, so another redirect issued later on would lead back to the origin page."),
+ *     ),
+ *   },
+ *   provides = {
+ *     "redirect" = @ContextDefinition("any",
+ *       label = @Translation("Redirect")
+ *     ),
+ *   }
+ * )
+ *
+ * @todo: Check if we can use context.url.type = "uri".
+ * @todo: Check if we can use context.force.restriction = "input".
+ * @todo: Check of we can use context.force.optional = true.
+ * @todo: Check of we can use context.force.defaultValue = true.
+ * @todo: Check if we can use context.destination.restriction = "input".
+ * @todo: Check of we can use context.destination.optional = true.
+ * @todo: Check of we can use context.destination.defaultValue = true.
+ *
+ */
+class SystemPageRedirect extends RulesActionBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The logger service.
+   *
+   * @var \Psr\Log\LoggerInterface
+   */
+  protected $logger;
+
+  /**
+   * @var \Drupal\Core\Routing\RedirectDestinationInterface
+   */
+  protected $redirectDestination;
+
+  /**
+   * Constructs a PageRedirect object.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Psr\Log\LoggerInterface $logger
+   *   The alias storage service..
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, LoggerInterface $logger, RedirectDestinationInterface $redirect_destination) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->logger = $logger;
+    $this->redirectDestination = $redirect_destination;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('logger.factory'),
+      $container->get('redirect.destination')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function summary() {
+    return $this->t('Page redirect');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function execute() {
+    $url = $this->getContextValue('url');
+    $force = $this->getContextValue('force');
+    $keepDestination = $this->getContextValue('destination');
+
+    // @todo: We need URL to Rules administration pages in order to make sure administrators won't get locked out from them.
+
+    // Make sure we do not redirect during batch processing.
+    $batch = batch_get();
+    if (isset($batch['current_set'])) {
+      $this->logger->log(LogLevel::WARNING, $this->t('Skipped page redirect during batch processing'));
+      $this->setProvidedValue('redirect', false);
```

Comment:
false, true, null should be upper case. FALSE, TRUE, NULL. Please check everywhere